torchio/data/subject.py

```diff
@@ -214,19 +214,46 @@ def apply_inverse_transform(self, **kwargs) -> 'Subject':
 
     def clear_history(self) -> None:
         self.applied_transforms = []
-
-    def check_consistent_attribute(self, attribute: str) -> None:
-        values_dict = {}
+        
+    def check_consistent_attribute(self,
+                               attribute: str,
+                               rtol=1e-6,
+                               atol=1e-6,
+                               message: Optional[str] = None) -> None:
         iterable = self.get_images_dict(intensity_only=False).items()
-        for image_name, image in iterable:
-            values_dict[image_name] = getattr(image, attribute)
-        num_unique_values = len(set(values_dict.values()))
-        if num_unique_values > 1:
+        if message is None:
             message = (
-                f'More than one {attribute} found in subject images:'
-                f'\n{pprint.pformat(values_dict)}'
-            )
-            raise RuntimeError(message)
+                f'More than one value for {attribute} found in subject images:'
+                '\n{}')
+        try:
+            first_attr = None
+            first_image = None
+
+            for image_name, image in iterable:
+                if first_attr is None:
+                    first_attr = getattr(image, attribute)
+                    first_image = image_name
+
+                else:
+                    curr_attr = getattr(image, attribute)
+                    if not np.allclose(curr_attr, first_attr, rtol=rtol,
+                                       atol=atol):
+                        message = message.format(
+                            pprint.pformat({
+                                first_image: first_attr,
+                                image_name: curr_attr
+                            }))
+                        raise RuntimeError(message)
+
+        except TypeError:
```

Comment:
This is currently not covered by the tests. When does this happen?

Reply:
Whenever something cannot be converted to a numpy numeric value. I guess that was on me, when I wrote a custom transform that had to check the orientation beforehand for example. And this method is now sufficiently generic, that it also allows that :)